### DISCUSSION TITLE
Version mismatch between CefSharp and Microsoft Edge WebView2 Runtime

### DISCUSSION THREAD

**Discussion by maaz-astera** [cefsharp/CefSharp#4947] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

128.4.90

### Operating System

Windows 10

### Architecture

x64

### .Net Version

.Net 8

### Implementation

WPF

### Reproduction Steps

I've added my project references like this:

```
<PackageReference Include="CefSharp.Wpf.NETCore" Version="128.4.90" />
	<PackageReference Include="chromiumembeddedframework.runtime" Version="128.4.90" />
	<PackageReference Include="chromiumembeddedframework.runtime.win-arm64" Version="128.4.90" />
	<PackageReference Include="chromiumembeddedframework.runtime.win-x64" Version="128.4.90" />
	<PackageReference Include="chromiumembeddedframework.runtime.win-x86" Version="128.4.90" />
```

My code to initialize CefSharp:

```
private static void LoadBrowserApp()
    {
        var settings = new CefSettings()
        {
            //@noloc
            CachePath = FilePath.Combine(AsteraAppContext.Paths.SystemFilePath, @"CefSharp\Cache"),
            LogSeverity = LogSeverity.Verbose,
            LogFile = FilePath.Combine(AsteraAppContext.Paths.SystemFilePath, "cefsharp.log")
        };
        //@noloc
        settings.CefCommandLineArgs.Add("enable-media-stream");
        //@noloc
        settings.CefCommandLineArgs.Add("disable-gpu", "1");
        //@noloc
        settings.CefCommandLineArgs.Add("disable-gpu-vsync", "1");
        //@noloc
        settings.CefCommandLineArgs.Add("disable-gpu-compositing", "1");
        CefSharpSettings.ShutdownOnExit = true;
        CefSharpSettings.SubprocessExitIfParentProcessClosed = true;
        if (Cef.Initialize(settings, performDependencyCheck: true, browserProcessHandler: null))
        {
            BrowserControlWPF.CefSharpSupport = true;
        }
        else
        {
            InfoBoxWPF.ShowError("CefSharp Browser failed to initialize.");
            BrowserControlWPF.CefSharpSupport = false;
        }
    }
```

	
	

### Expected behavior

CefSharp is supposed to initialize fine with no errors.

### Actual behavior

Running the project gives me CefSharp Browser failed to initialize error. 

Error: 
A procedure imported by 'C:\Users\username\source\repos\Project\Transfer\Client\bin\Debug\net8.0-windows\runtimes\win-x64\lib\netcoreapp3.1\CefSharp.Core.Runtime.dll' could not be loaded.

### Known Workarounds

Having been facing the issue for quite a while. I keep updating the CefSharp library version and it'd work fine but only for a few weeks. 
Now, it's in latest version and still facing this issue.

### Does this problem also occur in the CEF Sample Application

Yes using WPF/OffScreen command line args

### Other information

I've read a while ago that if there is a version mismatch between CefSharp and Microsoft Edge Webview2, it'd create issues.
Currently, the CefSharp version is 128.4.90 and WebView2 version is 129.0
Is there any way to fix this issue without keep updating my CefSharp version?

**Comment by amaitland** [comment]
Given `WebView2` is closed source, `MS` is probably the best place to debug this.

Others have reported similar issues at https://github.com/MicrosoftEdge/WebView2Feedback/issues/719

From the little I know, `WebView2` loads `Chromium` into a separate process. You'd have to ask them what version specific lib they're loading into your apps main process.

From a `CefSharp` point of view it's unlikely something I have control over. Remember that `CefSharp` is just a wrapper around the `Chromium Embedded Framework (CEF)`.

**Comment by Maazaowski** [comment]
I took a closer look and I think the issue isn’t related to a version mismatch between CefSharp and WebView2. It seems to be stemming from the Azure DevOps pipeline.

We build the release version of our WPF project through Azure DevOps. When we debug the project in the IDE (like Visual Studio or Rider), everything works smoothly, and CefSharp initializes correctly. However, when we run the release build compiled by Azure Pipelines, we encounter a 'CefSharp not initialized' error.

Upon further investigation, it appears that some Cef library files located in the path <project_folder>/runtimes/win-x64/native are missing from the release build, though they are included in the debug build.

I also tried creating a manual release build from my IDE, and that worked fine. It seems the problem lies with Azure DevOps.

I searched for solutions and found someone else experiencing the same issue on Microsoft's tech community, but there was no response.

Link: https://techcommunity.microsoft.com/t5/azure/compile-from-azure-pipeline-ignores-certain-files-related-to/m-p/3990780

Do you have any insights on what might be causing this? I’ve reviewed our pipelines multiple times but haven’t pinpointed the problem.
